neo4j-app/neo4j_app/core/elasticsearch/client.py

```diff
@@ -282,6 +282,10 @@ async def to_neo4j(
         # return the result summaries
         for task in neo4j_tasks:
             task.cancel()
+        # If there's nothing to import workers are cancelled right await, no need to
+        # wait for tasks to end
+        if not n_imported:
```

Comment:
I would have put these lines on line 279 closer to the declaration of `n_imported` to prevent misuse of the variable in between except if it's used by queue or neo4j_tasks but I don't see where...
Also there is a typo in the comment I think "right await" => "right away"

Reply:
Thank you

Reply:
Hum actually I still need to `neo4j_tasks` to be cancelled